exposure-notification/src/main/java/fi/thl/covid19/exposurenotification/batch/BatchFileService.java

```diff
@@ -78,9 +92,24 @@ public int cacheMissingBatchesBetween(int fromInterval, int untilInterval) {
         return batches.filter(previous::isBefore).collect(Collectors.toList());
     }
 
+    public List<BatchId> listBatchIdsSinceV2(BatchId previous, BatchIntervals intervals) {
+        Stream<BatchId> batches = dao.getAvailableIntervalsV2().stream()
+                .filter(i -> i != intervals.current && intervals.isDistributed(i))
+                .map(i -> new BatchId(fromV2to24hourInterval(i), Optional.of(i)));
+        if (intervals.current == intervals.last) {
+            batches = Stream.concat(batches, getDemoBatchIdV2(intervals.current).stream());
+        }
+        return batches.filter(previous::isBefore).collect(Collectors.toList());
+    }
+
     public Optional<BatchId> getDemoBatchId(int currentInterval) {
         int count = dao.getKeyCount(currentInterval);
-        return count > 0 ? Optional.of(new BatchId(currentInterval, Optional.of(count))) : Optional.empty();
+        return count > 0 ? Optional.of(new BatchId(currentInterval, Optional.of(Integer.parseInt("" + currentInterval + count)))) : Optional.empty();
```

Comment:
Seems a little odd to concatenate the numbers without a separator. Guess it works well enough for the demo feature, though. This is probably to avoid having to do firewall configuration changes for a different ID format?

Reply:
Yep, it's more like just to get it done in some manner. Reasoning behind this is more like that we don't entirely separate logic for demo batches. Anyway, good point and I'll check this in some point if it could be improved.